src/interface/TWAnyAddress.cpp

```diff
@@ -30,35 +24,31 @@ bool TWAnyAddressIsValid(TWString* _Nonnull string, enum TWCoinType coin) {
 struct TWAnyAddress* _Nullable TWAnyAddressCreateWithString(TWString* _Nonnull string,
                                                             enum TWCoinType coin) {
     const auto& address = *reinterpret_cast<const std::string*>(string);
-    auto normalized = TW::normalizeAddress(coin, address);
-    if (normalized.empty()) { return nullptr; }
-    return new TWAnyAddress{TWStringCreateWithUTF8Bytes(normalized.c_str()), coin};
+    auto *impl = TW::createAddress(address, coin);
+    if (impl == nullptr)
+        return nullptr;
+    return new TWAnyAddress{impl};
 }
 
 struct TWAnyAddress* _Nonnull TWAnyAddressCreateWithPublicKey(
     struct TWPublicKey* _Nonnull publicKey, enum TWCoinType coin) {
-    auto address = TW::deriveAddress(coin, publicKey->impl);
-    return new TWAnyAddress{TWStringCreateWithUTF8Bytes(address.c_str()), coin};
+    return new TWAnyAddress{TW::createAddress(publicKey->impl, coin)};
 }
 
 void TWAnyAddressDelete(struct TWAnyAddress* _Nonnull address) {
-    TWStringDelete(address->address);
+    delete address->impl;
```

Comment:
maybe we don't need this line now?

Reply:
It's a pointer allocated with "new". If we just destroy the object, nothing will free the pointer.